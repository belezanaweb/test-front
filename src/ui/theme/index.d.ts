import 'styled-components'

import { ITheme } from './types'

declare module 'styled-components' {
  export interface DefaultTheme extends ITheme { }
}
