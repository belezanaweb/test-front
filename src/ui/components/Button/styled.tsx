import styled from 'styled-components/macro'

export const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.primary};
  border: none;
  border-radius: 0.3rem;
  box-shadow: inset 0 -0.3rem 0 0 ${({ theme }) => theme.colors.primaryDark},
    0 0.2rem 0.4rem 0 rgba(0, 0, 0, 0.25);
  color: ${({ theme }) => theme.colors.baseLight};
  cursor: pointer;
  font-size: 2rem;
  font-weight: bold;
  letter-spacing: 0.05rem;
  line-height: 2.4rem;
  margin: auto;
  padding: 1.8rem 0;
  transition: background-color 0.3s ease;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
  &:hover {
    background-color: ${({ theme }) => theme.colors.primaryDark};
  }
  &:disabled {
    background-color: ${({ theme }) => theme.colors.base};
    box-shadow: none;
    cursor: not-allowed;
  }
`
