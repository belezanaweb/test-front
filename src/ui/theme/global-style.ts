import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: ${({ theme }) => theme.font};
  }

  html {
    font-size: 62.5%;
  }

  body {
    background-color: ${({ theme }) => theme.colors.secondaryDark};
    padding-bottom: 8rem;
  }

  img {
    max-width: 100%;
    display: block;
  }

  li {
    list-style: none;
  }
`

export default GlobalStyle
