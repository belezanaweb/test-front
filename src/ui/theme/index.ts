export default {
  font: 'Helvetica Neue, Helvetica, Arial, sans-serif',
  colors: {
    base: '#CCCCCC',
    baseAux: '#999999',
    baseDark: '#212122',
    baseLight: '#FFFFFF',
    error: '#FF3300',
    focus: '#A43287',
    primary: '#FF6C00',
    primaryLight: '#FF7800',
    primaryDark: '#D45A00',
    secondary: '#E7E7E7',
    secondaryLight: '#E0E7EE',
    secondaryDark: '#EEEEEE'
  }
}
