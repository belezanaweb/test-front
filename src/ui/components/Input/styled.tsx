import styled from 'styled-components/macro'

export const Input = styled.input<{error?: boolean}>`
  background-color: ${({ theme }) => theme.colors.baseLight};
  border-radius: 0.3rem;
  border: 0.1rem solid ${({ theme }) => theme.colors.secondary};
  box-shadow: inset 0 0.1rem 0.2rem 0 rgba(0, 0, 0, 0.2);
  padding: 1rem;
  color: ${({ theme }) => theme.colors.baseDark};
  transition: border 0.2s ease-out;
  width: 100%;
  &::placeholder {
    color: ${({ theme }) => theme.colors.secondaryLight};
  }
  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.focus};
    outline: 0;
  }
  ${({ theme, error }) => error && `border: 1px solid ${theme.colors.error};`}
`
