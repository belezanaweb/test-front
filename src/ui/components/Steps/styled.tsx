import styled from 'styled-components/macro'

import { IStyledStep } from './types'

export const Steps = styled.ul`
  display: flex;
  justify-content: space-between;
  padding: 1.2rem 2.5rem;
`

export const Step =
  styled.li <
  IStyledStep >
  `
  color: ${({ isActive, theme }) => (isActive ? theme.colors.primaryLight : theme.colors.base)};
  font-size: 1.3rem;
  font-weight: bold;
  transition: color 0.2s ease 0s;
  text-transform: uppercase;
`
