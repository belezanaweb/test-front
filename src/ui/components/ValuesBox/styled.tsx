import styled from 'styled-components/macro'

export const ValuesBox = styled.section`
  border: 1px solid ${({ theme }) => theme.colors.base};
  border-radius: 0.3rem;
  padding: 1.5rem;
  margin: 2rem 0;
`

export const ValuesBoxListItem = styled.p`
  color: ${({ theme }) => theme.colors.baseDark};
  display: flex;
  font-size: 1.4rem;
  justify-content: space-between;
  line-height: 1.7rem;
  margin-bottom: 0.8rem;
  text-transform: uppercase;
`

export const ValuesBoxActiveItem = styled(ValuesBoxListItem)`
  color: ${({ theme }) => theme.colors.primaryLight};
`

export const ValuesBoxHighlightedItem = styled(ValuesBoxListItem)`
  font-weight: bold;
  margin-bottom: 0;
  margin-top: 1.8rem;
`

export const ValuesBoxItem = styled.span``
