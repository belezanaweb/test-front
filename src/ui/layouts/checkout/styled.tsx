import styled from 'styled-components/macro'

export const Main = styled.main`
  padding: 0 1rem;

  @media (min-width: 360px) {
    padding: 0 2rem;
  }

  @media (min-width: 768px) {
    padding: 0 3rem;
  }
`
