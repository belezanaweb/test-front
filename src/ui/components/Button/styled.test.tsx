import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import * as S from './styled'

import theme from '../../theme'

describe('<Header /> styles', () => {
  it('<S.Header /> should have expected style rules', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <S.Button>Dummy content</S.Button>
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.primary)
    expect(container.firstChild).toHaveStyleRule('border', 'none')
    expect(container.firstChild).toHaveStyleRule('border-radius', '0.3rem')
    expect(container.firstChild).toHaveStyleRule(
      'box-shadow',
      `inset 0 -0.3rem 0 0 ${theme.colors.primaryDark}, 0 0.2rem 0.4rem 0`
    )
    expect(container.firstChild).toHaveStyleRule('color', theme.colors.baseLight)
    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer')
    expect(container.firstChild).toHaveStyleRule('font-size', '2rem')
    expect(container.firstChild).toHaveStyleRule('font-weight', 'bold')
    expect(container.firstChild).toHaveStyleRule('letter-spacing', '0.05rem')
    expect(container.firstChild).toHaveStyleRule('line-height', '2.4rem')
    expect(container.firstChild).toHaveStyleRule('margin', 'auto')
    expect(container.firstChild).toHaveStyleRule('padding', '1.8rem 0')
    expect(container.firstChild).toHaveStyleRule('transition', 'background-color 0.3s ease')
    expect(container.firstChild).toHaveStyleRule('text-align', 'center')
    expect(container.firstChild).toHaveStyleRule('text-decoration', 'none')
    expect(container.firstChild).toHaveStyleRule('text-transform', 'uppercase')
    expect(container.firstChild).toHaveStyleRule('width', '100%')

    // Hover
    expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.primaryDark, {
      modifier: ':hover'
    })

    // Disabled
    expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.base, {
      modifier: ':disabled'
    })
    expect(container.firstChild).toHaveStyleRule('box-shadow', 'none', {
      modifier: ':disabled'
    })
    expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed', {
      modifier: ':disabled'
    })
  })
})
