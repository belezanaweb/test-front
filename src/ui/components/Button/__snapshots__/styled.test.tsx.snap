// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> styles <S.Header /> should have expected style rules 1`] = `
.c0 {
  background-color: #FF6C00;
  border: none;
  border-radius: 0.3rem;
  box-shadow: inset 0 -0.3rem 0 0 #D45A00,0 0.2rem 0.4rem 0 rgba(0,0,0,0.25);
  color: #FFFFFF;
  cursor: pointer;
  font-size: 2rem;
  font-weight: bold;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
  line-height: 2.4rem;
  margin: auto;
  padding: 1.8rem 0;
  -webkit-transition: background-color 0.3s ease;
  transition: background-color 0.3s ease;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
}

.c0:hover {
  background-color: #D45A00;
}

.c0:disabled {
  background-color: #CCCCCC;
  box-shadow: none;
  cursor: not-allowed;
}

<button
  class="c0"
>
  Dummy content
</button>
`;
